integration/tests/identity_provider/identity_provider_test.go

```diff
@@ -0,0 +1,379 @@
+)
+
+const (
+	oidcGroupName = "oidc-reader"
+)
+
+var (
+	params                  *tests.Params
+	oidcConfig              *OIDCConfig
+	cleanupCognitoResources func() error
+)
+
+type OIDCConfig struct {
+	clientID     string
+	idToken      string
+	refreshToken string
+	idpIssuerURL string
+}
+
+func init() {
+	// Call testing.Init() prior to tests.NewParams(), as otherwise -test.* will not be recognised. See also: https://golang.org/doc/go1.13#testing
+	testing.Init()
+	params = tests.NewParams("identity-provider")
+}
+
+func TestIdentityProvider(t *testing.T) {
+	testutils.RegisterAndRun(t)
+}
+
+var _ = BeforeSuite(func() {
+	if !params.SkipCreate {
+		cmd := params.EksctlCreateCmd.WithArgs(
+			"cluster",
+			"--verbose", "4",
+			"--name", params.ClusterName,
+			"--kubeconfig", params.KubeconfigPath,
+			"--nodes", "1",
+		)
+		Expect(cmd).To(RunSuccessfully())
+	}
+
+	var err error
+	fmt.Fprintf(GinkgoWriter, "creating Cognito OIDC provider\n")
+	oidcConfig, err = setupCognitoProvider(params.ClusterName, params.Region)
+	Expect(err).NotTo(HaveOccurred())
+	fmt.Fprintf(GinkgoWriter, "created Cognito provider; client ID: %s\n", oidcConfig.clientID)
+})
+
+var _ = Describe("(Integration) [Identity Provider]", func() {
+
+	It("should associate, get and disassociate identity provider", func() {
+		By("associating a new identity provider")
+		identityProviderClusterConfig := makeIdentityProviderClusterConfig(oidcConfig, params.ClusterName, params.Region)
+
+		cmd := params.EksctlCmd.WithArgs(
+			"associate",
+			"identityprovider",
+			"--config-file", "-",
+			"--verbose", "4",
+			"--wait",
+		).
+			WithStdin(strings.NewReader(identityProviderClusterConfig)).
+			WithoutArg("--region", params.Region).
+			WithTimeout(40 * time.Minute)
+
+		Expect(cmd).To(RunSuccessfully())
+
+		By("getting the identity provider")
+		cmd = params.EksctlGetCmd.
+			WithArgs(
+				"identityprovider",
+				"--cluster", params.ClusterName,
+				"-o", "yaml",
+			)
+		Expect(cmd).To(RunSuccessfullyWithOutputStringLines(
+			ContainElement(ContainSubstring(fmt.Sprintf("ClientID: %s", oidcConfig.clientID)))),
+		)
+
+		By("creating RBAC resources")
+		test, err := kube.NewTest(params.KubeconfigPath)
+		Expect(err).NotTo(HaveOccurred())
+		defer test.Close()
+
+		test.CreateClusterRoleFromFile("testdata/cluster-role.yaml")
+		test.CreateClusterRoleBindingFromFile("testdata/cluster-role-binding.yaml")
+
+		By("creating an OIDC Clientset")
+		clientset, err := createOIDCClientset(eks.New(NewSession(params.Region)), oidcConfig, params.ClusterName)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("reading Kubernetes resources")
+		list, err := clientset.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{})
+		Expect(err).NotTo(HaveOccurred())
+		Expect(list.Items).To(HaveLen(1))
+
+		secrets, err := clientset.CoreV1().Secrets(metav1.NamespaceAll).List(context.TODO(), metav1.ListOptions{})
+		Expect(err).NotTo(HaveOccurred())
+		Expect(secrets.Items).NotTo(BeEmpty())
```

Comment:
Can we actually verify what those Items contain?

Reply:
The contents of the items are irrelevant here. The client code mainly needs to check that it has access to secret resources in all namespaces.